src/containers/AccountProvider/index.tsx

```diff
@@ -82,13 +91,16 @@ export const AccountProvider = ({ children }: AccountProviderProps) => {
     );
   }
 
+  const availableRoles = [...new Set(filterNodes(data?.viewer?.groups?.nodes?.map((group) => filterNodes(group?.permissions)).flat()))];
```

Comment:
This implementation works as intended but the value will be recalculated on every render of a component that depends on `usePermissions` hook. 

The first thing I would suggest is wrapping it in an `useMemo` and recalculating only when `data?.groups` changes. 

The second suggestion is to use a Map or a simple object where the key is the specific role and the value is `true`. This way in the role check in the `usePermission` hook or the `PermissionGuard` the complexity would be 0(1) instead of O(n). In the permission check instead of `availableRoles.includes` we can just call `availableRoles[targetRole]`.

Such performance optimizations are optional but I thought it might be suitable for a functionality that is being executed all of the time.